File: C_Programming/Practise_Programs/PP_043.c
/*
==============================================================================
 Program     : Display Negative Numbers from -N to 0

 Description : Reads an integer from the user and displays all numbers from
               the negative of the specified value up to 0.

 Author      : Shubham Gadhe
==============================================================================
*/  

#include<stdio.h>

/*
==============================================================================
 Function    : Display

 Description : Displays all numbers from -N to 0.

 Parameters  : iNo - Absolute value used to generate the sequence.

 Returns     : None

 Complexity  : Time  : O(n)
               Space : O(1)
==============================================================================
*/
void Display(int iNo)
{
    int iCnt = 0;

    // Display numbers from -N to 0.
    for(iCnt = -iNo; iCnt <= 0; iCnt++)
    {
        printf("%d\t", iCnt);
    }
    
    printf("\n");

}

int main()
{
    int iValue = 0;

    printf("Enter number : \n");
    scanf("%d", &iValue);

    Display(iValue);

    return 0;
}

/*
==============================================================================
 Sample Input
------------------------------------------------------------------------------
 5

 Sample Output
------------------------------------------------------------------------------
 -5    -4    -3    -2    -1    0
==============================================================================
*/

File: Java_Programming/Practise_Programs/PP_043.java
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_043
//
//  Description       : This program accepts a number from the user
//                      and displays odd numbers while replacing
//                      even numbers with '*'.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 06/08/2026
//
//  Time Complexity   : O(n)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

import java.util.*;

class PP_043
{
    //////////////////////////////////////////////////////////////////
    //
    //  Function Name    : Display()
    //
    //  Description      : It is used to display odd numbers
    //                     from 1 to the specified number.
    //                     Even numbers are replaced with '*'.
    //
    //  Author           : Shubham Somanath Gadhe
    //  Date             : 06/08/2026
    //
    //////////////////////////////////////////////////////////////////

    public static void Display(int iNo)
    {
        int iCnt = 0;

        for(iCnt = 1; iCnt <= iNo; iCnt++)
        {
            if(iCnt % 2 == 0)
            {
                System.out.print("*\t");
            }
            else
            {
                System.out.print(iCnt + "\t");
            }
        }

        System.out.println();
    }

    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        int iValue = 0;

        System.out.println("Enter the number of elements : ");
        iValue = sobj.nextInt();

        Display(iValue);
    }
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input  :
//      5
//
//  Sample Output :
//      1    *    3    *    5
//
//////////////////////////////////////////////////////////////////